lib/metanorma/collection/collection.rb

```diff
@@ -0,0 +1,228 @@
+# frozen_string_literal: true
+
+require "relaton"
+require "relaton/cli"
+require "metanorma-utils"
+require_relative "../util/util"
+require_relative "../util/disambig_files"
+require_relative "../collectionconfig/collectionconfig"
+
+module Metanorma
+  class FileNotFoundException < StandardError; end
+
+  class AdocFileNotFoundException < StandardError; end
+
+  # Metanorma collection of documents
+  class Collection
+    attr_reader :file
+
+    # @return [Array<String>] documents-inline to inject the XML into
+    #   the collection manifest; documents-external to keeps them outside
+    attr_accessor :directives, :documents, :bibdatas, :coverpage, :dirname
+    attr_accessor :disambig, :manifest, :bibdata, :compile
+
+    # @param file [String] path to source file
+    # @param dirname [String] directory of source file
+    # @param directives [Array<String>] documents-inline to inject the XML into
+    #   the collection manifest; documents-external to keeps them outside
+    # @param bibdata [RelatonBib::BibliographicItem]
+    # @param manifest [Metanorma::CollectionManifest]
+    # @param documents [Hash<String, Metanorma::Document>]
+    # @param prefatory [String]
+    # @param coverpage [String]
+    # @param final [String]
+    def initialize(**args)
+      @file = args[:file]
+      @dirname = File.expand_path(File.dirname(@file)) # feeds @manifest
+      @documents = args[:documents] || {} # feeds initialize_directives
+      @bibdatas = args[:documents] || {}
+      initialize_vars
+      initialize_config(args[:config])
+      initialize_directives
+      @documents.merge! @manifest.documents(@dirname)
+      @bibdatas.merge! @manifest.documents(@dirname)
```

Comment:
There isn need to pass `@dirname` inside because you can call `@dirname` inside the `documents` method.

Reply:
No, it is called recursively within manifest on submanifests. That's how I get YAML in different directories to point to the right files when iterated.